src/ui/components/Toolbar.tsx

```diff
@@ -159,6 +138,32 @@ export default function Toolbar({
     }
   }, [selectedPrimaryPanel, showCommentsBadge]);
 
+  const togglePanel = () => {
+    const panel = sidePanelRef.current;
+    if (panel) {
+      if (sidePanelCollapsed) {
+        panel.expand();
+      } else {
+        panel.collapse();
+      }
+    }
+  };
+
+  const handleButtonClick = (panelName: PrimaryPanelName) => {
+    const samePanelSelected = selectedPrimaryPanel === panelName;
+    const shouldTogglePanel = sidePanelCollapsed || samePanelSelected;
+
+    if (!samePanelSelected) {
+      trackEvent(`toolbox.primary.${panelName}_select`);
+      dispatch(actions.setSelectedPrimaryPanel(panelName));
+    }
+
+    if (shouldTogglePanel) {
+      trackEvent(`toolbox.toggle_sidebar`);
+      togglePanel();
+    }
```

Comment:
Sorry if this is a silly question, but why are we checking this conditional before toggling? Shouldn't we always toggle expanded/collapsed when a panel header is clicked?

Reply:
This isn't panel headers, this is the sidebar buttons.

If you go from, say, "Info" to "Pause", and the sidebar was already open, then there's nothing to toggle (which is part of the trickiness here because we've got imperative "toggle" behavior rather than "ensure this is open/closed" logic).

Reply:
Oh right. Thanks for the reminder. 😄 Reviewing with bad connectivity (so I'm not trying the code out like I normally would) and I got those mixed up. My bad.